### PR TITLE
DEVOPS-3714: Adding own collector

### PR DESCRIPTION
**Pull request by tkocemba**

### CODE REVIEW

File: appdynamics/resource_collector.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+func createCollector(d *schema.ResourceData) client.Collector {
+
+	collector := client.Collector{
+		Name:      d.Get("name").(string),
+		Type:      d.Get("type").(string),
+		Hostname:  d.Get("hostname").(string),
+		Port:      d.Get("port").(int),
+		Username:  d.Get("username").(string),
+		Password:  d.Get("password").(string),
+		AgentName: d.Get("agent_name").(string),
```
Comment (waihungshum-wr) [inline appdynamics/resource_collector.go:87]: need to add enabled field

File: appdynamics/resource_collector.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+func updateCollector(d *schema.ResourceData, collector client.Collector) {
+	d.Set("name", collector.Name)
+	d.Set("type", collector.Type)
+	d.Set("hostname", collector.Hostname)
+	d.Set("port", collector.Port)
+	d.Set("username", collector.Username)
+	// Password is always set as `appdynamics_redacted_password` so we need to always overwrite this
+	//d.Set("password", collector.Password)
+	d.Set("agent_name", collector.AgentName)
```
Comment (waihungshum-wr) [inline appdynamics/resource_collector.go:154]: Do we need to set enabled value here?

File: example/collectors/main.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+resource appdynamics_collector test2 {
+	name="example1"
+	type="MYSQL"
+	hostname="test2"
+	username="u"
+	password="password2"
+	port=17
+	agent_name="test"
```
Comment (waihungshum-wr) [inline example/collectors/main.tf:26]: enabled?

File: appdynamics/resource_action_test.go [hunk 4]
```diff
@@ -298,3 +298,46 @@ func configureConfig() string {
 					  default = %s
 					}`, os.Getenv("APPD_SECRET"), os.Getenv("APPD_CONTROLLER_BASE_URL"), os.Getenv("APPD_SCOPE_ID"), os.Getenv("APPD_APPLICATION_ID"))
 }
+
```
Comment (waihungshum-wr) [inline appdynamics/resource_action_test.go:301]: should this go in the resource_collector_test.go file instead?

**Review by waihungshum-wr — COMMENTED** [review]
a few comments

File: appdynamics/resource_collector.go (hunk 1 above)
Comment (tkocemba) [reply appdynamics/resource_collector.go:87]: fixed

File: appdynamics/resource_collector.go (hunk 2 above)
Comment (tkocemba) [reply appdynamics/resource_collector.go:154]: fixed

File: example/collectors/main.tf (hunk 3 above)
Comment (tkocemba) [reply example/collectors/main.tf]: solved

**Review by waihungshum-wr — APPROVED** [review]
LGTM

**Pull request merged**